src/main/java/org/reactome/qa/doisuggester/ReactionlikeEvent.java

```diff
@@ -0,0 +1,354 @@
+			if (!found)
+			{
+				newInstanceEdits.add(instanceEdit);
+			}
+		}
+		return newInstanceEdits;
+	}
+
+	/**
+	 * Filters a list of InstanceEdits such that only "non-Reactome" InstanceEdits are returned.
+	 * And InstanceEdits is considered to be "non-Reactome" if at least one author of the InstanceEdit is not
+	 * a member of the Reactome project.
+	 * @param instanceEdits
+	 * @return
+	 */
+	public static List<GKInstance> filterForNonReactomeInstanceEdits(List<GKInstance> instanceEdits)
+	{
+		List<GKInstance> nonReactomeInstanceEdits = new ArrayList<>();
+
+		try
+		{
+			// For an InstanceEdit to be considered non-Reactome, there must exist at least one author whose project is
+			// not "Reactome"
+			for (GKInstance instanceEdit : instanceEdits)
+			{
+				List<GKInstance> authors = instanceEdit.getAttributeValuesList(ReactomeJavaConstants.author);
+				if (authors != null && !authors.isEmpty())
+				{
+					int i = 0;
+					boolean isNonReactome = false;
+					while (!isNonReactome && i < authors.size())
+					{
+						GKInstance author = authors.get(i);
+						// TODO: Notify Guanming that "project" is not in ReactomeJavaConstants.
+						String project = (String) author.getAttributeValue("project");
+						if (project == null || project.trim().equals("") || !project.trim().equalsIgnoreCase("reactome") )
+						{
+							isNonReactome = true;
+							nonReactomeInstanceEdits.add(instanceEdit);
+						}
+						i++;
+					}
+				}
+//				else
+//				{
+//					// Log a warning about an InstanceEdit with NO authors - there's probably some data cleanup that needs to happen.
+//				}
+			}
+		}
+		catch (InvalidAttributeException e)
+		{
+			e.printStackTrace();
+		}
+		catch (Exception e)
+		{
+			e.printStackTrace();
+		}
+		return nonReactomeInstanceEdits;
+	}
+
+	/**
+	 * Gets the furthest ancestors of this Reaction-like event which does not have newInstanceEdit as one of its InstanceEdits.
+	 * This method will perform a recursive search from a reaction-like event up to a top-level pathway, searching for a pathway that
+	 * does not have a "new" InstanceEdit. A set of all such pathways is returned.
+	 * @param newInstanceEdit a "new" InstanceEdit.
+	 * @return The pathway objects.
+	 */
+	public Set<GKInstance> getFurthestAncestorsWithoutNewInstanceEdit(GKInstance newInstanceEdit)
+	{
+		List<GKInstance> parents;
+		try
+		{
+			this.ancestors.addAll(this.getFurthestAncestorWithoutNewInstanceEdit(this.getUnderlyingInstance(), newInstanceEdit));
+		}
+		catch (Exception e)
+		{
+			// TODO Auto-generated catch block
+			e.printStackTrace();
+		}
+		return this.ancestors;
+	}
+
+	/**
+	 * Private recursive method to search up the Pathway hierarchy.
+	 * This method gets the ancestors of a pathway/RLE that is furthest up the tree and also does not have the same InstanceEdit.
+	 * @param A pathway to examine.
+	 * @param newInstanceEdit the InstanceEdit to check for.
+	 * @return
```

Comment:
Could this @return have the same description of the method it overrides?

Reply:
This method does not override anything...

Reply:
Oh, did you mean "overload"?

Reply:
Yeah, sorry - I did mean overload.